memote/support/basic.py

```diff
@@ -104,11 +106,38 @@ def find_ngam(model):
           http://doi.org/10.1038/nprot.2009.203
 
     """
-    atp_adp_conv_rxns = helpers.find_converting_reactions(model, ("atp", "adp"))
-    return [rxn for rxn in atp_adp_conv_rxns
-            if rxn.build_reaction_string() == 'atp_c + h2o_c --> '
-                                              'adp_c + h_c + pi_c' and not
-            rxn.lower_bound <= 0]
+    atp_adp_conv_rxns = helpers.find_converting_reactions(
+        model, ("MNXM3", "MNXM7")
+    )
+    id_of_main_compartment = helpers.find_compartment_id_in_model(model, 'c')
+
+    reactants = {
+        helpers.find_met_in_model(model, "MNXM3", id_of_main_compartment)[0],
+        helpers.find_met_in_model(model, "MNXM2", id_of_main_compartment)[0]
+    }
+
+    products = {
+        helpers.find_met_in_model(model, "MNXM7", id_of_main_compartment)[0],
+        helpers.find_met_in_model(model, "MNXM1", id_of_main_compartment)[0],
+        helpers.find_met_in_model(model, "MNXM9", id_of_main_compartment)[0]
+    }
+
+    candidates = [rxn for rxn in atp_adp_conv_rxns
+                  if set(rxn.reactants) == reactants and
+                  set(rxn.products) == products and
+                  rxn.reversibility is False]
```

Comment:
If you move the last reversibility check to the first position it will help the performance of this test since the set based conditions might unnecessary.